docs/CppInteroperability/GettingStartedWithCppInterop.md

```diff
@@ -0,0 +1,209 @@
+#  Getting started with C++ Interoperability
+
+This document is desgined to get you started with bidirectional API-level interoperability between Swift and C++.
+
+## Table of Contents
+
+- [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code)
+- [Adding C++ to an Xcode project](#adding-c-to-an-xcode-project)
+- [Creating a Swift Package](#Creating-a-Swift-Package)
+- [Building with CMake](#building-with-cmake)
+
+## Creating a Module to contain your C++ source code
+
+- Create a new C++ implementation and header file, and generate a bridging header when prompted
+- For this example we will call the files Cxx, so we should have a Cxx.cpp and Cxx.hpp.
+- Import your C++ header, into the bridging header 
+```
+//In ProjectName-Bridging-Header
+#import "Cxx.hpp"
+```
+Create a module.modulemap file to expose your source code 
+- In order to expose our C++ code we must designate a separate as the source.
+- Create an empty file and call it `module.modulemap`
+- In this file create the module for your source code, and define your C++ header
+```
+//In module.modulemap
+module Cxx {
+//note that your header should be the file that containts your method implementations
+    header "Cxx.cpp
```

Comment:
Nit: `requires cplusplus` isn't required but it's convention for C++ modules, especially if they use C++ features.

Reply:
oh okay, thank you for the clarification